pycon.md

```diff
@@ -398,59 +441,80 @@ at some point.
+~~non-existant~~ not always accurate. 
 
-You might all know PyPI, that's usually where most of you may download your
-packages from. Pip does interrogate PyPI to install packages. PyPI is old. PyPI
-testing is sparse (understand non existant), and documentation is... ~~non
-existant~~ not alway accurate, so not that easy to run locally.
+As a result, it's not easy to run PyPI locally.
 
 -- 
 
-The PyPA have stated developing Warehouse (the new PyPI), which is well
-documented, 100% test coverage and provided with a one liner to run it locally
+The PyPA stated developing Warehouse (the new, improved PyPI), which is well
+documented, with 100% test coverage. It even has a one liner to run it locally
 using Docker. 
 
 -- 
 
-In production PyPI and warehouse are connected to the same Postgres database.
-So any update need to be coordinated. 
+In production, PyPI and warehouse are connected to the same Postgres database.  
+So any updates need to be coördinated between them. 
 
 -- 
 
-Seem pretty easy, when you requires `/simple/<package>` webpage the sql query
-should simply:
+It seems like it should be easy… 
 
-    SELECT * from release_files where package_name=package
+When you need the `/simple/<package>` webpage the sql query should simply be:
 
-And build a list of href. Seem like a solved Problem right ? Except reading the
-actual pep teh requires-python are per **release** not per file (ie you can't
-have a wheel which is python 3.3+ and a sdist 3.2+)
+```sql
+SELECT * from release_files where package_name=package
+```
 
--- 
+And build a list of href. 
 
-Release is a different table, so we could "simply" use a JOIN. Except with the
-number of available packages the join is too slow. We can't really refactor
-the all database format because PyPI is not really tested.
+Problem solved, right? 
 
-We ended up using a postgress trigger with UPSERT (that simplify some logic a
-lot) that update the `release_files` table as soon as it (or `release`) get
-updated.  
+-- 
+
+Unforutnately PEP 345 specifies that `requires-python` is an attribute 
+on **releases**, not **release files**.
 
-We've improved the documentation of warehouse, and PyPI, so now you should be
-able to contribute back more easily. Despite not being well tested there is a
-number of low hanging fruit ! Either cleaning up the codebase, or bringing
-features from PyPI to Warehouse.
+I.e., you can't have a wheel which is python 3.3+ and a sdist 3.2+.
 
+↑ TODO: Matthias, I don't follow this point… why would you want wheels vs sdists 
+from the same release to be different?
 
 -- 
 
-## Setuptools
+Because `release` is a different table from `release_files`, in theory, 
+we could use a `JOIN` for the two tables. 
 
-We did not do the patches, ut it allow you to set metadata
+Except, with the number of available packages, a join is too slow. 
 
+On the other hand, we cannot safely refactor the database because PyPI is not well tested.
+
+--
 
+#### Solution: A trigger on UPSERT
```

Comment:
Triggering an "UPSERT" is not really a solution. The fact that we can use upsert is convenient. It just avoids complex conditional logic.